docker-cpp-demos/amazonlinux-docker/Dockerfile

```diff
@@ -0,0 +1,53 @@
+# Build docker with
+# docker build -t kinesis-video-producer-sdk-cpp-amazon-linux .
+#
+FROM amazonlinux:latest
+RUN yum update -y && \
+	yum upgrade -y && \
+	yum install -y \
+	autoconf \
+	automake  \
+	bison \
+	bzip2 \
+	cmake \
+	curl \
+	diffutils \
+	flex \
+	gcc \
+	gcc-c++ \
+	git \
+	gmp-devel \
+	gstreamer* \
+	libffi \
+	libffi-devel \
+	libmpc-devel \
+	libtool \
+	make \
+	m4 \
+	mpfr-devel \
+	pkgconfig \
+	vim \
+	wget \
+	xz
+
+RUN yum install -y libcurl-devel 
+
+RUN wget https://cmake.org/files/v3.12/cmake-3.12.3.tar.gz && \
+    tar zxvf cmake-3.* && \
+    rm cmake-3.12.3.tar.gz
+
+WORKDIR cmake-3.12.3/
+RUN ./bootstrap
+RUN make -j4
+RUN make install
+
+WORKDIR /opt/
```

Comment:
I would combine all of these build steps into a single `RUN` and use `cd` instead of `WORKDIR`. Every time you use `RUN`, it'll create an extra layer which will increase the bundle size, https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers.

But, obviously, feel free to ignore this comment if you think the bundle size doesn't really matter.